solidity/test/AcreMultiAssetVault.test.ts

```diff
@@ -0,0 +1,949 @@
+          )
+          .withArgs(await thirdParty.getAddress())
+      })
+    })
+
+    context("when the caller is the governance", () => {
+      context("when the asset is not supported", () => {
+        it("should revert", async () => {
+          const notSupportedAsset =
+            await ethers.Wallet.createRandom().getAddress()
+
+          await expect(
+            multiAssetVault
+              .connect(governance)
+              .removeSupportedAsset(notSupportedAsset),
+          ).to.be.revertedWithCustomError(multiAssetVault, "AssetNotSupported")
+        })
+      })
+
+      context("when the asset is supported", () => {
+        beforeAfterSnapshotWrapper()
+
+        let tx: ContractTransactionResponse
+
+        before(async () => {
+          tx = await multiAssetVault
+            .connect(governance)
+            .removeSupportedAsset(await asset1.getAddress())
+        })
+
+        it("should emit SupportedAssetRemoved event", async () => {
+          await expect(tx)
+            .to.emit(multiAssetVault, "SupportedAssetRemoved")
+            .withArgs(await asset1.getAddress())
+        })
+
+        it("should remove the asset from the supported assets", async () => {
+          expect(
+            await multiAssetVault.supportedAssets(await asset1.getAddress()),
+          ).to.be.false
+        })
+      })
+    })
+  })
+
+  describe("deposit", () => {
+    beforeAfterSnapshotWrapper()
+
+    let caller: HardhatEthersSigner
+    let depositAmount: bigint
+
+    let initialDepositCount: bigint
+    let expectedDepositId: bigint
+
+    let callResult: bigint
+    let tx: ContractTransactionResponse
+
+    before(async () => {
+      caller = depositor1
+      depositAmount = to1e18(10)
+
+      initialDepositCount = await multiAssetVault.depositCount()
+      expectedDepositId = initialDepositCount + 1n
+
+      await asset1.mint(await caller.getAddress(), depositAmount)
+      await asset1
+        .connect(caller)
+        .approve(await multiAssetVault.getAddress(), depositAmount)
+
+      callResult = await multiAssetVault
+        .connect(caller)
+        .deposit.staticCall(await asset1.getAddress(), depositAmount)
+      tx = await multiAssetVault
+        .connect(caller)
+        .deposit(await asset1.getAddress(), depositAmount)
+    })
+
+    it("should return the deposit id", () => {
+      expect(callResult).to.be.equal(expectedDepositId)
+    })
+
+    it("should increment deposit count", async () => {
+      expect(await multiAssetVault.depositCount()).to.be.equal(
+        initialDepositCount + 1n,
+      )
+    })
+
+    it("should emit DepositCreated event", async () => {
+      await expect(tx)
+        .to.emit(multiAssetVault, "DepositCreated")
+        .withArgs(
+          await caller.getAddress(),
+          await asset1.getAddress(),
+          expectedDepositId,
+          depositAmount,
+        )
+    })
+
+    it("should transfer the asset to the vault", async () => {
```

Comment:
Not really to the vault but to the Mezo Portal looking at `changeTokenBalances` assertion.

Reply:
6bd6ff4df9709343674e7f6c96c2283b7fbcc80d